x-pack/plugins/observability_solution/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -50,7 +55,8 @@ export type MetricThresholdAlert = Omit<
   'kibana.alert.evaluation.values'
 > & {
   // Defining a custom type for this because the schema generation script doesn't allow explicit null values
-  'kibana.alert.evaluation.values'?: Array<number | null>;
+  [ALERT_EVALUATION_VALUES]?: Array<number | null>;
+  [ALERT_EVALUATION_THRESHOLD]?: Array<number | null>;
```

Comment:
@maryam-saeidi I think we should consider persisting range values (for `in between` condition) even if it is not possible to display them in alert table at the moment. We can figure out later how we can use these values in UI. If we want to add these later in alert document, it could be a breaking change, since we are saving threshold as number right now and for `in between` it would be an array of number.

Reply:
It is an array of numbers even if we only have one threshold, so we will not have any breaking change in this case.

I checked the ES Query rule, and they also don't save the multi-threshold comparators such as `in between`, so I decided to keep it consistent between rules. If we decide to save this information, then it would be better to do it for all the rules together. (including ES query rule)

Reply:
The array of numbers is currently for the multiple conditions, isn't it? If there are 3 conditions in the rule, there will be 3 thresholds stored in this array. If there is a range threshold in a condition, wouldn't we need `Array<number | number[] | null>`?

Reply:
perhaps there was no plan to use these values in UI for ES query rule, but since we are making use of these values, I think we should improve this for the rules we are already working on.

Reply:
> The array of numbers is currently for the multiple conditions, isn't it? 

Yes, it is for multiple conditions.

> If there are 3 conditions in the rule, there will be 3 thresholds stored in this array. If there is a range threshold in a condition, wouldn't we need Array<number | number[] | null>?

I think `Array<number | null>` is the current structure that we consider for this field in ES, I don't know if ES also covers `Array<number | number[] | null>` or if it only accepts a number or an array of numbers for the first level. Do you have any idea about it?

> perhaps there was no plan to use these values in UI for ES query rule, but since we are making use of these values, I think we should improve this for the rules we are already working on.

Since we use ES Query in observability, I think we can come up with a unified way and propose the same to the ResponseOps team or even do it ourselves similar to what we want to apply for other observability rules. In the end, from the users' perspective, there is no difference in which team is responsible for a rule, they see all of them as observability rule types.

Reply:
> In the end, from the users' perspective, there is no difference in which team is responsible for a rule, they see all of them as observability rule types.

I agree, I just meant since we are already working on infra rules in this PR, we can provide complete solution including range condition.

I checked the overwriting of alert documents with values containing `number[]`, and it didn't cause any issue/error. The mapping has `scaled_float` type and I suppose it treats array/non-array same way as long as type (`number` in this case) is same. Ideally I would prefer to include range thresholds in same PR, but since it won't be a breaking change, I will leave it upto you to include now or later together with other rules.

Reply:
https://github.com/elastic/kibana/issues/178067